main.py

```diff
@@ -0,0 +1,258 @@
+"""
+This main python file contains the backend logic of the website, including
+the reception of Cloud Pub/Sub messages and communication with GCS and datastore.
+"""
+
+# Import all necessary libraries.
+import webapp2
+import jinja2
+import os
+import logging
+import json
+import urllib
+import collections
+import cloudstorage as gcs
+import googleapiclient.discovery
+from google.appengine.ext import ndb
+from google.appengine.ext import blobstore
+from google.appengine.api import images
+
+# Set up jinja2 for HTML templating.
+template_dir = os.path.join(os.path.dirname(__file__), 'templates')
+jinja_environment = jinja2.Environment(loader = jinja2.FileSystemLoader(template_dir))
+
+# Constants. Note: Change THUMBNAIL_BUCKET and PHOTO_BUCKET to
+# be applicable to your project.
+THUMBNAIL_BUCKET = 'thumbnails-bucket'
+PHOTO_BUCKET = 'shared-photo-album'
+NUM_NOTIFICATIONS_TO_DISPLAY = 50
+MAX_LABELS = 5
+
+# Describes a Notification to be displayed on the home/news feed page.
```

Comment:
You can use a docstring instead of a comment. (this is application is a couple of other cases here too)